internal/server/http/handler/v1/treecluster/handler.go

```diff
@@ -28,12 +28,30 @@ var (
 // @Router			/v1/cluster [get]
 // @Param			page	query	string	false	"Page"
 // @Param			limit	query	string	false	"Limit"
-// @Param			status	query	string	false	"Status"
 // @Security		Keycloak
 func GetAllTreeClusters(svc service.TreeClusterService) fiber.Handler {
 	return func(c *fiber.Ctx) error {
 		ctx := c.Context()
-		domainData, err := svc.GetAll(ctx)
+
+		pageParam, err := strconv.Atoi(c.Query("page", "1"))
+		if err != nil || pageParam < 1 {
+			err := service.NewError(service.BadRequest, "invalid page format")
+			return errorhandler.HandleError(err)
+		}
+		page := int32(pageParam)
+
+		limit := int32(-1)
+		limitParam := c.Query("limit", "-1")
+		if limitParam != "" {
+			limitParam, err := strconv.Atoi(limitParam)
+			if err != nil || (limitParam != -1 && limitParam <= 0) {
+				err := service.NewError(service.BadRequest, "invalid limit format")
+				return errorhandler.HandleError(err)
+			}
+			limit = int32(limitParam)
+		}
+
+		domainData, totalCount, err := svc.GetAll(ctx, page, limit)
```

Comment:
I would love to see this in the middleware. So we can use it with every entity in our app. Here is a small blog post on how to implement pagination in the middleware. This is an example for gin, but you can easily translate it to fibers middleware.

https://articles.wesionary.team/make-pagination-easy-in-golang-using-pagination-middleware-and-gorm-scope-a5f6eb3bebaa